frontend/src/config/index.ts

```diff
@@ -1 +1 @@
-export const baseApiUrl = 'http://localhost:8080';
+export const baseApiUrl = import.meta.env.VITE_BASE_API_URL;
```

Comment:
Please add for better debugging experience.
```suggestion
if (!import.meta.env.VITE_BASE_API_URL) {
  throw 'VITE_BASE_API_URL no set';
}

export const baseApiUrl = import.meta.env.VITE_BASE_API_URL;
```

Or we could use a library like tiny-invariant. Makes for prettier code I think.
```ts
import invariant from 'tiny-invariant';

invariant(!import.meta.env.VITE_BASE_API_URL, 'VITE_BASE_API_URL no set');
```

https://www.npmjs.com/package/tiny-invariant

Reply:
Better be careful with (tiny) libs/packages. If really needed we should give this a decision, maybe something like https://github.com/jaredpalmer/tsdx#invariant is better suited?

Reply:
I agree with not throwing too many packages at the project.
However, I would go with the defensive route of checking that this is set.
Not only is it good for developer experience but also when deploying to make sure this is set.
However, throwing a string literal seems like bad practice since catching it seems messy.
Maybe throw a subclass of `Error`, e.g. `ConfigurationError`?

Reply:
>However, throwing a string literal seems like bad practice since catching it would be messy.
Maybe throw a subclass of Error, e.g. ConfigurationError?

The way it is setup currently, we would not catch errors from the config, because they happen before the app renders for the first time. But I agree, if we would ever want to, throwing an `Error` would be more sane, sorry for that 😅

Reply:
I think the suggestion is better than adding a tiny lib just for a single function call. Reminds me a bit of the [leftpad story](https://www.davidhaney.io/npm-left-pad-have-we-forgotten-how-to-program/).

+1 for throwing an error instead of a string :)

Reply:
I am using `import.meta.env.hasOwnProperty` now, so that `VITE_BASE_API_URL` can be an empty string without failing.